src/services/rooms.service.js

```
@@ -0,0 +1,64 @@
+import { Room } from '../models/room.js';
+
+const getAllRooms = async () => {
+  return Room.findAll({
+    order: [['createdAt', 'DESC']],
+  });
+};
+
+const createRoom = async (title, userId, description = '') => {
+  if (!title || !userId) {
+    throw new Error('Title and userId are required');
+  }
+
+  return Room.create({ title, userId, description });
+};
+
+const deleteRoom = async (id) => {
+  if (!id) {
+    throw new Error('Room id is required');
+  }
+
+  const deletedCount = await Room.destroy({ where: { id } });
+
+  if (deletedCount === 0) {
+    throw new Error('Room not found');
+  }
+
+  return deletedCount;
```

When deleting, you throw 'Room not found' if no rows were deleted which is correct, but consider using a specific error type so controllers can map it to 404. Also think about related data: if messages belong to a room, ensure database FK constraints or cascade delete behavior are defined so deleting a room doesn't leave orphan messages or silently fail elsewhere.